plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java

```diff
@@ -143,7 +143,7 @@ public Map<SchemaTableName, ConnectorViewDefinition> getViews(Optional<String> s
                 getView(name).ifPresent(view -> views.put(name, view));
             }
             catch (TrinoException e) {
-                if (e.getErrorCode().equals(TABLE_NOT_FOUND.toErrorCode())) {
+                if (e.getErrorCode().equals(TABLE_NOT_FOUND.toErrorCode()) || e instanceof TableNotFoundException || e instanceof ViewNotFoundException) {
```

Comment:
why checking exception instance. Can you look check if error code is either `TABLE_NOT_FOUND` or `NOT_FOUND`.

Also can we unify? Are there many places which still throw exception with `TABLE_NOT_FOUND`?

Reply:
generally NOT_FOUND feels too broad (eg ColumnNotFoundException, SchemaNotFoundException, ...)

> Also can we unify?

this idk